zerver/tests/test_upload.py

```diff
@@ -62,10 +75,45 @@ def test_file_upload_authed(self):
         base = '/user_uploads/'
         self.assertEquals(base, uri[:len(base)])
 
+        # In the future, local file requests will follow the same style as S3
+        # requests; they will be first authenthicated and redirected
         response = self.client.get(uri)
         data = "".join(response.streaming_content)
         self.assertEquals("zulip!", data)
 
+        # Check if DB has attachment marked as unclaimed
+        entry = Attachment.objects.get(file_name='zulip.txt')
+        self.assertEquals(entry.is_claimed(), False)
+
+    def test_delete_old_unclaimed_attachments(self):
+        # Create some dummy file entries which are older than a week
+        sender_email = "hamlet@zulip.com"
+        user_profile = get_user_profile_by_email(sender_email)
+        two_week_ago = timezone.now() - datetime.timedelta(weeks=2)
+        dummy_1 = Attachment.objects.create(
+                                    file_name='zulip.txt',
+                                    path_id='1/31/4CBjtTLYZhk66pZrF8hnYGwc/zulip.txt',
+                                    owner=user_profile,
+                                    create_time=two_week_ago
+                                    )
+        dummy_2 = Attachment.objects.create(
+                                    file_name='temp_file.py',
+                                    path_id='1/31/4CBjtTLYZhk66pZrF8hnYGwc/temp_file.py',
+                                    owner=user_profile,
+                                    create_time=two_week_ago
+                                    )
+
+        # Send message refering only dummy_1
+        self.subscribe_to_stream(sender_email, "Denmark")
+        body = "Some files here ...[zulip.txt](http://localhost:9991/user_uploads/1/31/4CBjtTLYZhk66pZrF8hnYGwc/zulip.txt)"
+        self.send_message(sender_email, "Denmark", Recipient.STREAM, body, "test")
+
+        # Get list of old unclaimed atachments
+        weeks_ago = 2
+        old_attachments = get_old_unclaimed_attachments(weeks_ago)
+        self.assertIn(dummy_2, old_attachments)
+        self.assertTrue(dummy_1 not in old_attachments)
```

Comment:
No, we should test the deletion flow end-to-end.  Otherwise, we might break the implementation in some future refactoring.  For most features that'd be fine, but because the risk of bugs here is data either being incorrectly deleted or incorrectly not deleted, it's important to test the entire code flow.